PurchasesCoreSwiftTests/SubscriberAttributes/AttributionDataMigratorTests.swift

```diff
@@ -0,0 +1,640 @@
+        expect(converted.count) != 0
+        checkCommonAttributes(in: converted)
+        let expectedMapping = [
+            SpecialSubscriberAttributes.adjustID: AttributionKey.Adjust.id.rawValue,
+            SpecialSubscriberAttributes.mediaSource: AttributionKey.Adjust.network.rawValue,
+            SpecialSubscriberAttributes.campaign: AttributionKey.Adjust.campaign.rawValue,
+            SpecialSubscriberAttributes.adGroup: AttributionKey.Adjust.adGroup.rawValue,
+            SpecialSubscriberAttributes.creative: AttributionKey.Adjust.creative.rawValue
+        ]
+        checkConvertedAttributes(converted: converted, original: adjustData, expectedMapping: expectedMapping)
+    }
+
+    func testAdjustAttributionConversionGivesPreferenceToAdIdOverRCNetworkID() {
+        let adjustData = adjustData(addAdId: true, addNetworkID: true)
+        let converted = attributionDataMigrator.convertAttributionDataToSubscriberAttributes(
+                attributionData: adjustData, network: AttributionNetwork.RCAttributionNetworkAdjust)
+        expect(converted.count) != 0
+        checkCommonAttributes(in: converted)
+        let expectedMapping = [
+            SpecialSubscriberAttributes.adjustID: AttributionKey.Adjust.id.rawValue,
+            SpecialSubscriberAttributes.mediaSource: AttributionKey.Adjust.network.rawValue,
+            SpecialSubscriberAttributes.campaign: AttributionKey.Adjust.campaign.rawValue,
+            SpecialSubscriberAttributes.adGroup: AttributionKey.Adjust.adGroup.rawValue,
+            SpecialSubscriberAttributes.creative: AttributionKey.Adjust.creative.rawValue
+        ]
+        checkConvertedAttributes(converted: converted, original: adjustData, expectedMapping: expectedMapping)
+    }
+
+    func testAdjustAttributionConversionConvertsRCNetworkIDCorrectly() {
+        let adjustData = adjustData(addAdId: false, addNetworkID: true)
+        let converted = attributionDataMigrator.convertAttributionDataToSubscriberAttributes(
+                attributionData: adjustData, network: AttributionNetwork.RCAttributionNetworkAdjust)
+        expect(converted.count) != 0
+        checkCommonAttributes(in: converted)
+        let expectedMapping = [
+            SpecialSubscriberAttributes.adjustID: AttributionKey.networkID.rawValue,
+            SpecialSubscriberAttributes.mediaSource: AttributionKey.Adjust.network.rawValue,
+            SpecialSubscriberAttributes.campaign: AttributionKey.Adjust.campaign.rawValue,
+            SpecialSubscriberAttributes.adGroup: AttributionKey.Adjust.adGroup.rawValue,
+            SpecialSubscriberAttributes.creative: AttributionKey.Adjust.creative.rawValue
+        ]
+        checkConvertedAttributes(converted: converted, original: adjustData, expectedMapping: expectedMapping)
+    }
+
+    func testAdjustAttributionConversionWorksWithNullIDFA() {
+        let adjustData = adjustData(withIDFA: nil)
```

Comment:
These tests are failing because I am inserting nils in the dictionary, which I am actually not sure it represents the real use of our addAttributionData function. I've looked at our swift interface and the data we accept in `Purchases.addAttributionData` is of type `[AnyHashable: Any]`, but in these tests I assume developers can pass nil as values in the data. I don't think there's a way developers can do that right? @aboedo tagging you since you reviewed the Android counterpart and might know the answer.

Reply:
tricky one. 

in objc, you can't set `nil` in a dict, but you _can_ do `NSNull`. 
in swift, you can't pass in something that's `nil` directly, as it won't technically be `Any` if it's nil. 
So the following code:

```swift
Purchases.addAttributionData(["foo": nil], from: .adjust)
```

produces 
```
'nil' is not compatible with expected dictionary value type 'Any'
```

buuuut
the following code: 

```swift
let myDict: [String: Any?] = ["lalala": nil]
Purchases.addAttributionData(myDict, from: .adjust)
```

compiles just fine, because `Any?` counts as `Any`. 
As you have probably guessed, using `Any` sucks. We just always strive to constrain the interface further than that to prevent this kind of issue. 

I tried this out just to see what would happen with the old implementation, and it seems to get translated to `NSNull`.

Reply:
and the NSNull eventually gets sent to the backend as the string `"<null>"` I believe. So in order to preserve this behavior we just need to make sure that we're still compatible with something like 

```swift
Purchases.addAttributionData(["foo": NSNull()], from: .adjust)
```